UIKitDZ/UIKitDZ/MessageViewController.swift

```diff
@@ -0,0 +1,128 @@
+        textField.font = UIFont.boldSystemFont(ofSize: 18)
+        textField.textAlignment = .center
+        return textField
+    }()
+
+    private let sendAgainButton = {
+        let button = UIButton(frame: .init(x: 109, y: 288, width: 167, height: 36))
+        button.setTitle("Отправить снова", for: .normal)
+        button.setTitleColor(
+            UIColor(red: 89.0 / 250.0, green: 190.0 / 250.0, blue: 199.0 / 250.0, alpha: 1.0),
+            for: .normal
+        )
+        button.backgroundColor = UIColor(red: 239.0 / 250.0, green: 247.0 / 250.0, blue: 247.0 / 250.0, alpha: 1.0)
+        button.layer.cornerRadius = 10
+        return button
+    }()
+
+    private let confirmButton = {
+        let button = UIButton(frame: .init(x: 20, y: 632, width: 335, height: 44))
+        button.isEnabled = false
+        button.setTitle("Подтвердить", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.backgroundColor = UIColor(red: 89.0 / 250.0, green: 190.0 / 250.0, blue: 199.0 / 250.0, alpha: 0.3)
+        button.layer.cornerRadius = 12
+        return button
+    }()
+
+    // MARK: - Life Cycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+
+        codeTextField.addTarget(self, action: #selector(editingChanged), for: .editingChanged)
+        confirmButton.addTarget(self, action: #selector(confirmTapped), for: .touchUpInside)
+        backButton.addTarget(self, action: #selector(backTapped), for: .touchUpInside)
+    }
+
+    // MARK: - Private Methods
+
+    private func setupUI() {
+        view.backgroundColor = .white
+        view.addSubview(confirmButton)
+        view.addSubview(smsCodeLabel)
+        view.addSubview(enterCodeLabel)
+        view.addSubview(codeView)
+        view.addSubview(codeTextField)
+        view.addSubview(sendAgainButton)
+        view.addSubview(backButton)
+        // navigationItem.leftBarButtonItem = UIBarButtonItem(customView: backButton)
+    }
+
+    @objc private func editingChanged() {
+        guard let code = codeTextField.text,
+              !code.isEmpty
+        else {
+            confirmButton.isEnabled = false
+            confirmButton.backgroundColor = UIColor(
+                red: 89.0 / 250.0,
+                green: 190.0 / 250.0,
+                blue: 199.0 / 250.0,
+                alpha: 0.3
+            )
+            return
+        }
+        confirmButton.isEnabled = true
+        confirmButton.backgroundColor = UIColor(
+            red: 89.0 / 250.0,
+            green: 190.0 / 250.0,
+            blue: 199.0 / 250.0,
+            alpha: 1.0
+        )
+    }
+
+    @objc private func confirmTapped() {
+        print("confirm tapped")
+        let thanksVC = ThanksViewController()
+        thanksVC.modalPresentationStyle = .fullScreen
+        present(thanksVC, animated: true)
+    }
+
+    @objc func backTapped() {
+        print("кнопка баск нажмакана")
+    }
```

Comment:
```suggestion
    @objc func backTapped() { }
```
 и зачем метод без реализации